mne/epochs.py

```diff
@@ -1826,8 +1827,12 @@ def __init__(self, data, info, events, tmin=0, event_id=None,
         if data.shape[0] != len(events):
             raise ValueError('The number of epochs and the number of events'
                              'must match')
-        tmax = (data.shape[2] - 1) / info['sfreq'] + tmin
-        if event_id is None:  # convert to int to make typing-checks happy
+        index_offset = 1 if data.shape[2] > 1 else 0
+        tmax = (data.shape[2] - index_offset) / info['sfreq'] + tmin
+        if index_offset == 0:
+            reject_tmin, reject_tmax = tmin, tmax
+        if event_id is None or isinstance(event_id, int):
```

Comment:
these should be separate conditionals, then:
```
if event_id is None
    ...
elif isinstance(event_id, int):
    event_id = dict((str(event_id), event_id)
```
or so. But again, this should probably be handled by the `_BaseEpochs` class, not `EpochsArray` since you can pass single integers as the `event_id` to many `Epochs` classes, and they should all be handled in the same place (DRY).